mainMARS.s

```diff
@@ -217,12 +217,57 @@ max:
   pop_ra_and_return
 
 # Power operation.
+.data 
+power_a_message: .asciiz "Enter the a in (a ^ b):\n"
+power_b_message: .asciiz "Enter the b in (a ^ b):\n"
+zero_exponent_message: .asciiz "The result equals 1 " 		#Special_Case if exponent equals zero,Then the result equals one
 
 .text
-power:
-  push_ra
-  la REG_PRINT_STRING_ARG, unimplemented_message
+power:  
+  # Print the message for a.
+  la REG_PRINT_STRING_ARG, power_a_message
+  jal printString
+
+  # Read a.
+  jal readInteger
+  move $t1, REG_READ_INTEGER_RET
```

Comment:
Don't store values in temporary registers if you will be calling functions after. Those registers are not callee-saved, meaning their values may change across function calls. Use saved registers instead.

Reply:
You mean that i would use **$s0 - $s7**

Reply:
@MohamedAbdallah17 Yes.